services/src/main/java/org/keycloak/services/resources/IdentityBrokerService.java

```diff
@@ -974,6 +974,8 @@ private Response performAccountLinking(AuthenticationSessionModel authSession, U
             userSession.setNote(Details.IDENTITY_PROVIDER_USERNAME, context.getUsername());
         }
 
+        updateFederatedIdentity(context, federatedUser);
```

Comment:
@toddkazakov I think calling this method here will cause errors when linking existing accounts. Could you please try to add the call at https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/services/resources/IdentityBrokerService.java#L942?